src/libs/Notification/PushNotification/shouldShowPushNotification.ts

```diff
@@ -14,12 +14,17 @@ export default function shouldShowPushNotification(pushPayload: PushPayload): bo
 
     // The payload is string encoded on Android
     if (typeof payload === 'string') {
-        payload = JSON.parse(payload) as string;
+        try {
+            payload = JSON.parse(payload) as JsonObject;
+        } catch {
+            Log.hmmm(`[PushNotification] Failed to parse the payload`, payload);
+            payload = undefined;
+        }
```

Comment:
We repeated this payload parsing logic in a few places. Can we move this to a helper function `parsePushNotificationPayload` to DRY this up? You can just put that in the [PushNotification lib](https://github.com/Expensify/App/blob/5a2602c441170093e99850e2194808218b0aa838/src/libs/Notification/PushNotification/index.native.ts) if it doesn't cause any dependency cycles.